Sources/Subprocess/AsyncBufferSequence.swift

```diff
@@ -74,6 +80,239 @@ public struct AsyncBufferSequence: AsyncSequence, Sendable {
+        public typealias Element = String
+
+        private let base: AsyncBufferSequence
+        private let bufferingPolicy: BufferingPolicy
+
+        public struct AsyncIterator: AsyncIteratorProtocol {
+            public typealias Element = String
+
+            private var source: AsyncBufferSequence.AsyncIterator
+            private var buffer: [Encoding.CodeUnit]
+            private var eofReached: Bool
+            private var startIndex: Int
+            private let bufferingPolicy: BufferingPolicy
+
+            internal init(
+                underlyingIterator: AsyncBufferSequence.AsyncIterator,
+                bufferingPolicy: BufferingPolicy
+            ) {
+                self.source = underlyingIterator
+                self.buffer = []
+                self.eofReached = false
+                self.startIndex = 0
+                self.bufferingPolicy = bufferingPolicy
+            }
+
+            public mutating func next() async throws -> String? {
+
+                func loadBuffer() async throws -> [Encoding.CodeUnit]? {
+                    guard !self.eofReached else {
+                        return nil
+                    }
+
+                    guard let buffer = try await self.source.next() else {
+                        self.eofReached = true
+                        return nil
+                    }
+                    #if os(Windows)
+                    // Cast data to CodeUnit type
+                    let result = buffer.withUnsafeBytes { ptr in
+                        return Array(
+                            UnsafeBufferPointer<Encoding.CodeUnit>(
+                                start: ptr.bindMemory(to: Encoding.CodeUnit.self).baseAddress!,
+                                count: ptr.count / MemoryLayout<Encoding.CodeUnit>.size
+                            )
+                        )
+                    }
+                    #else
+                    // Unfortunately here we _have to_ copy the bytes out because
+                    // DisptachIO (rightfully) reuses buffer, which means `buffer.data`
+                    // has the same address on all iterations, therefore we can't directly
+                    // create the result array from buffer.data
+                    let temporary = UnsafeMutableBufferPointer<Encoding.CodeUnit>.allocate(
+                        capacity: buffer.data.count
+                    )
+                    defer { temporary.deallocate() }
+                    let actualBytesCopied = buffer.data.copyBytes(
+                        to: temporary,
+                        count: buffer.data.count
+                    )
+
+                    // Calculate how many CodePoint elements we have
+                    let elementCount = actualBytesCopied / MemoryLayout<Encoding.CodeUnit>.stride
+
+                    // Create array by copying from the buffer reinterpreted as CodePoint
+                    let result: Array<Encoding.CodeUnit> = Array(
```

Comment:
Isn't this two copies? Can we create the array directly from the buffer instead?